sopt/src/main/scala/dagr/sopt/util/ParsingUtil.scala

```diff
@@ -0,0 +1,141 @@
+
+  /** Initializes the color for printing */
+  private def initializeColor(color: String): String = {
+    if (printColor) color
+    else ""
+  }
+
+  /** Provides ANSI colors for the terminal output **/
+  val KNRM: String = initializeColor("\u001B[0m")
+  //val KBLD: String = initializeColor("\u001B[1m")
+  val KRED: String = initializeColor("\u001B[31m")
+  val KGRN: String = initializeColor("\u001B[32m")
+  val KYEL: String = initializeColor("\u001B[33m")
+  val KBLU: String = initializeColor("\u001B[34m")
+  val KMAG: String = initializeColor("\u001B[35m")
+  val KCYN: String = initializeColor("\u001B[36m")
+  val KWHT: String = initializeColor("\u001B[37m")
+  val KBLDRED: String = initializeColor("\u001B[1m\u001B[31m")
+  val KERROR: String = initializeColor("\u001B[1m\u001B[31m")
+  //val KERROR: String = initializeColor("\u001B[35;7;1m") // BLINKING: "\u001B[35;5;1m"
+
+  /** Gets a mapping between sub-classes of [[T]] and their [[clp]] annotation.
+    *
+    * @param srcClasses the source classes from which to build the map.
+    * @param omitSubClassesOf a list of omit subclasses to omit from the returned map.
+    * @param includeHidden include classes whose [[clp]] annotation has `hidden` set to true.
+    * @tparam T the super class.
+    * @return a mapping between sub-classes of [[T]] and their [[clp]] annotation.
+    */
+  private def classToPropertyMapFromSourceClasses[T](srcClasses: Traversable[Class[_]],
+                                                     omitSubClassesOf: Iterable[Class[_]] = Nil,
+                                                     includeHidden: Boolean = false)
+  : Map[Class[_ <: T], clp] = {
+
+    // Filter out interfaces, synthetic, primitive, local, or abstract classes.
+    def keepClass(clazz: Class[_]): Boolean = {
+      !clazz.isInterface && !clazz.isSynthetic && !clazz.isPrimitive && !clazz.isLocalClass && !Modifier.isAbstract(clazz.getModifiers)
+    }
+
+    // Find all classes with the annotation
+    val classes: Traversable[Class[T]] = srcClasses
+      .filter { keepClass }
+      .filterNot { clazz => omitSubClassesOf.exists { _.isAssignableFrom(clazz) } }
+      .filter {
+        findClpAnnotation(_) match {
+          case None      => false
+          case Some(clp) => includeHidden || !clp.hidden
+        }
+      }.map(_.asInstanceOf[Class[T]])
+
+    // Get all the name collisions
+    val nameCollisions = classes
+      .groupBy(_.getSimpleName)
+      .filter { case (name, cs) => cs.size > 1 }
+      .map    { case (name, cs) => cs.mkString(", ") }
+
+    // SimpleName should be unique
+    if (nameCollisions.nonEmpty) {
+      throw new CommandLineException(s"Simple class name collision: ${nameCollisions.mkString(", ")}")
+    }
+
+    // Finally, make the map
+    classes.map(clazz => Tuple2(clazz, ReflectionUtil.findJavaAnnotation(clazz, classOf[clp]).get)).toMap
+  }
+
+
+  /** Gets a mapping between sub-classes of [[T]] and their properties.
```

Comment:
The comment is a bit out of date since this is where the actual searching for the classes happens.